care/facility/models/facility.py

```diff
@@ -289,6 +298,39 @@ def get_features_display(self):
     CSV_MAKE_PRETTY = {"facility_type": (lambda x: REVERSE_FACILITY_TYPES[x])}
 
 
+class FacilityHubSpoke(BaseModel):
+    hub = models.ForeignKey(Facility, on_delete=models.CASCADE, related_name="hub_set")
+    spoke = models.ForeignKey(
+        Facility, on_delete=models.CASCADE, related_name="spoke_set"
+    )
+    relationship = models.IntegerField(
+        choices=HubRelationship.choices, default=HubRelationship.REGULAR_HUB
+    )
+
+    class Meta:
+        constraints = [
+            # Ensure hub and spoke are not the same
+            CheckConstraint(
+                check=~models.Q(hub=models.F("spoke")),
+                name="hub_and_spoke_not_same",
+            ),
+            # bidirectional uniqueness
```

Comment:
This ensures bi directional uniqueness, but it does not make sense to have cyclical dependences

Reply:
I think the check is already there in serializers if I am understanding you correctly. I have also added it in the model now. Can you please verify if that is what you meant?